File: coordinatorbehavior/src/main/java/com/kirchhoff/coordinatebehavior/CoordinatorBehaviorActivity.java
package com.kirchhoff.coordinatebehavior;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

public class CoordinatorBehaviorActivity extends AppCompatActivity {


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.a_coordinate_behavior);

        findViewById(R.id.bApply).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Button with custom behavior",
                        Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });

        findViewById(R.id.fab).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Button with custom behavior",
                        Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });
    }
}

File: coordinatorbehavior/src/main/java/com/kirchhoff/coordinatebehavior/CoordinatorBehaviorActivity.kt
package com.kirchhoff.coordinatebehavior

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.snackbar.Snackbar

class CoordinatorBehaviorActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.a_coordinate_behavior)

        findViewById<View>(R.id.bApply).setOnClickListener { view -> showSnackbar(view, "Button with custom behavior") }
        findViewById<View>(R.id.fab).setOnClickListener { view -> showSnackbar(view, "FAB with custom behavior") }
    }

    private fun showSnackbar(view: View, text: String) {
        Snackbar.make(view, text, Snackbar.LENGTH_LONG).setAction("Action", null).show()
    }
}
